File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/infrastructure/persistence/repositories/JpaTenantIpAllowlistRepository.java
package br.com.deltaglobalbank.identity.infrastructure.persistence.repositories;

import java.util.List;
import java.util.UUID;

import br.com.deltaglobalbank.identity.infrastructure.persistence.entities.TenantIpAllowlistEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface JpaTenantIpAllowlistRepository extends JpaRepository<TenantIpAllowlistEntity, UUID> {
    List<TenantIpAllowlistEntity> findAllByTenantId(UUID tenantId);

    boolean existsByCidrAndTenantId(String cidr, UUID tenantId);
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/infrastructure/persistence/repositories/JpaTenantIpAllowlistRepository.kt
package br.com.deltaglobalbank.identity.infrastructure.persistence.repositories

import br.com.deltaglobalbank.identity.infrastructure.persistence.entities.TenantIpAllowlistEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface JpaTenantIpAllowlistRepository : JpaRepository<TenantIpAllowlistEntity, UUID> {
    fun findAllByTenantId(tenantId: UUID): List<TenantIpAllowlistEntity>
    fun existsByCidrAndTenantId(cidr: String, tenantId: UUID): Boolean
}